ingest/Maybe.py

```diff
@@ -0,0 +1,55 @@
+__author__ = 'Stephan'
+
+from itertools import chain
+
+class Maybe:
+    def __init__(self, v=None):
+        self.value = v
+
+    @staticmethod
+    def nothing():
+        return Maybe()
+
+    @staticmethod
+    def of(t):
```

Comment:
Truthfully, creating shared python classes, functions, etc... should have been a separate branch and not tied to the dataset browser. That way we could go ahead and push the dataset browser and worked on the other thing for all ingest scripts not just datasets.